library/core/src/iter/traits/collect.rs

```diff
@@ -4,9 +4,11 @@
 /// created from an iterator. This is common for types which describe a
 /// collection of some kind.
 ///
-/// [`FromIterator::from_iter()`] is rarely called explicitly, and is instead
-/// used through [`Iterator::collect()`] method. See [`Iterator::collect()`]'s
-/// documentation for more examples.
+/// If you want to create a collection from the contents of an iterator, the
+/// [`Iterator::collect()`] method is preferred. However, the compiler is
+/// sometimes unable to infer the full type of the collection. In these cases,
```

Comment:
Rather than "unable", maybe this can talk about turbofish?  Since there's a good reason it doesn't, rather than it being a gap in functionality.

First stab at it, please improve:

> However, in places where you need to specify the container type, `FromIterator` can sometimes be more convenient than using a turbofish (like `::<Vec<_>>()`).

Reply:
This is the part from the RFC I was working from:

> it is reasonably common that type inferencing fails to infer the full type of the target type, in which case an explicit type annotation or turbofishing is needed

It's probably better just to refer to the syntax. Type annotations good syntax, but I think people choose turbofishing for the sake of consistency since most variable declarations don't need a type annotation.

```
If you want to create a collection from the contents of an iterator, the
[`Iterator::collect()`] method is preferred. However, when you need to
specify the container type, [`FromIterator::from_iter()`] can be more 
readable than using a turbofish (e.g. `::<Vec<_>>()`). See the
[`Iterator::collect()`] documentation for more examples of its use.
```